intro-to-R-tidyverse/04c-intro_to_tidyverse_exercise-part-2.Rmd

```diff
@@ -0,0 +1,170 @@
+output:
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+tidyverse exercises - part 2
+
+The goal of these exercises is to practice the visualization techniques and 
+data frame manipulations we used in [02-intro_to_ggplot2](./01-intro_to_ggplot2.Rmd) 
+and [03-intro_to_tidyverse](./01-intro_to_tidyverse.Rmd) notebooks. 
+
+Here will use another [microrarray dataset](https://www.refine.bio/experiments/GSE19578/integrated-molecular-genetic-profiling-of-pediatric-high-grade-gliomas-reveals-key-differences-with-adult-disease) that is [pre-processed for us by refine.bio](http://docs.refine.bio/en/latest/main_text.html#processing-information).
+This study from [Paugh et al, 2010](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2903336/) has glioblastoma samples from a range of patient ages. 
+
+### Set Up 
+
+In this notebook, we will be using tidyverse packages. 
+You can load the `library()` or use the `::` strategy. 
+
+```{r}
+
+```
+
+It's convenient to declare our directory names at the beginning so we can change
+it once and have it change throughout the notebook. 
+Remember we need to be consistent with what we used in the previous notebook. 
+
+```{r}
+data_dir <- 
+plots_dir <- 
+```
+
+## Read in the cleaned metadata
+
+Use `readr::read_tsv()` to read in the filtered gene expression matrix TSV, 
+"filtered_GSE19578.tsv" and assign it the name `genes_df`. 
+
+```{r}
+
+```
+
+Use `readr::read_tsv()` to read in the cleaned metadata TSV, "GSE19578.tsv" 
+and assign it the name `metadata`. 
+
+```{r}
+
+```
+
+## Set up grouped summary statistics for genes
+
+Let's calculate some fold changes for our groups. 
+We've set up a data frame that is transposed so you can use using `tidyverse` functions. 
+
+```{r}
+transposed_genes_df <- genes_df %>% 
+  # We want the gene information to be stored as rownames for the next set of 
+  # steps we are doing
+  tibble::column_to_rownames("Gene") %>% 
+  # Transposing the data frame so it is sample x genes
+  t() %>% 
+  # t() function turns our data frame into a matrix so let's turn it back to 
+  # a data.frame so we can use it with tidyverse functions
+  as.data.frame() %>% 
+  # By default our sample ids will be stored as rownames, let's bring it out as 
+  # its own columns 
+  tibble::rownames_to_column("sample_id")
+```
+
+Use `dplyr::select()` and an `%in%` statement to choose ~2-5 genes of interest. 
+Can use [GeneCard](https://www.genecards.org/) to find the corresponding Ensembl
+ID to your favorite genes. 
+If you don't have genes in particular you are interested in, feel free to use 
+these: `ENSG00000134323` (MYCN), `ENSG00000124762`(CDKN1A), and 
+`ENSG00000115414` (FN1).
+
+```{r}
+genes_oi <- c("ENSG00000134323", 
+              "ENSG00000124762", 
+              "ENSG00000115414")
+
+```
+
+Use `dplyr::inner_join()` to join the `metadata_clean` columns to 
+`transposed_genes_df` by the `sample_id`. 
+
+```{r}
+
+```
+
+Calculate the group means of your `select()`ed genes. 
```

Comment:
Does the word "group" here refer to the age group? Is left open intentionally, in order to allow participants to choose their grouping variable?

Reply:
I will be more specific!

Reply:
But I guess they *could* use any variable.

Reply:
But some work better than others, so I was more explicit here too.